drools-wb-screens/drools-wb-drl-text-editor/drools-wb-drl-text-editor-backend/pom.xml

```diff
@@ -143,6 +143,17 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+      <scope>test</scope>
+    </dependency>
```

Comment:
There are few rule and some exceptions. From what I was able to observe there are three main rules to follow:
 * always use SLF4J API (slf4j-api jar) for the logging calls (not log4j, Java Util Logging, or Commons Logging)
 * use Logback as logging implementation (logback-classic) where it makes sense (testing, end-user apps/bundles)
 * exclude other logging implementations coming transitively. These are usually commons-logging and log4j. Lot's of 3rd party deps are dragging logging implementations with them. That is generally bad and should be fixed on the library side. But until that happens we need to exclude them ourselves. Of course nothing is that easy, so whenever you exclude the logging impl, you should make sure that there is a logging bridge defined so that the logging actually works for those 3rd party libraries. See http://www.slf4j.org/legacy.html which nicely explains these bridges. So for example whenever you exclude log4j, you should add dependency on log4j-over-slf4j (unless it's already there).

Only "end-user" applications, like jcr2vfs migration tool or kie-config-cli tool should actually depend on the logging implementation. One exception are the WAR files which should _not_ ideally contain any logging implementation and we should just rely on the logging impl bundled within the container. Other modules should only use it with test scope, so it is not included in the transitive dependency list. 

We should remove any dependency on log4j and any log4j.xml files. 

Huh, I am not sure the explanation above is clear enough as it is hard to describe some of these rules and exceptions. Hopefully I haven't made this more complicated than it already was :)